napari/layers/_labels_layer/model.py

```diff
@@ -28,51 +31,88 @@ class Labels(Layer):
         Image metadata.
     multichannel : bool, optional
         Whether the image is multichannel. Guesses if None.
+    opacity : float, optional
+        Opacity of the labels, must be between 0 and 1.
     name : str, keyword-only
         Name of the layer.
     num_colors : int, optional
         Number of unique colors to use. Default used if not given.
     **kwargs : dict
         Parameters that will be translated to metadata.
     """
-    def __init__(self, label_image, meta=None, *, name=None, num_colors=256, **kwargs):
+    def __init__(self, label_image, meta=None, *, name=None, num_colors=256,
+                 opacity=0.7, **kwargs):
         if name is None and meta is not None:
             if 'name' in meta:
                 name = meta['name']
 
         visual = ImageNode(None, method='auto')
         super().__init__(visual, name)
-        self.events.add(colormap=Event)
+        self.events.add(colormap=Event, mode=Event, n_dimensional=Event,
+                        contiguous=Event, brush_size=Event,
+                        selected_label=Event)
 
         self.seed = 0.5
         self._raw_image = label_image
-        self._max_label = np.max(label_image)
-        self._image = colormaps._low_discrepancy_image(self._raw_image, self.seed)
+        self._image = self.raw_to_displayed(self._raw_image)
         self._meta = meta
         self.interpolation = 'nearest'
         self.colormap_name = 'random'
         self.colormap = colormaps.label_colormap(num_colors)
-        self.opacity = 0.7
 
+        self._node.opacity = opacity
+        self._n_dimensional = True
+        self._contiguous = True
+        self._brush_size = 10
+        self._last_cursor_coord = None
+
+        self._selected_label = 0
+        self._selected_color = None
+
+        self._mode = Mode.PAN_ZOOM
+        self._mode_history = self._mode
+        self._status = str(self._mode)
+        self._help = 'enter paint or fill mode to edit labels'
 
         # update flags
         self._need_display_update = False
         self._need_visual_update = False
 
         self._qt_properties = QtLabelsLayer(self)
+        self._qt_controls = QtLabelsControls(self)
 
         self._node.clim = [0., 1.]
         self.events.colormap()
 
+    def raw_to_displayed(self, raw):
+        """Determines displayed image from a saved raw image and a saved seed.
+        This function ensures that the 0 label gets mapped to the 0 displayed
+        pixel
+
+        Parameters
+        -------
+        raw : array | float
+            Raw input image
+
+        Returns
+        -------
+        image : array
+            Image mapped between 0 and 1 to be displayed
+        """
+        image = np.where(raw > 0, colormaps._low_discrepancy_image(raw,
+                         self.seed), 0)
+        return image
```

Comment:
Have you tried debugging here and checking whether 0 labels are mapped to the value 0?

Reply:
Or you could print `image[raw == 0].max()` and see what horrors come out. =P

Reply:
I get 0

Reply:
Another idea: add the value of _image to the status bar

Reply:
`_image` is definitely 0.0 when the label is 0. The color that is shown corresponds to the color of `colormap([0.0000001])` though